src/redux/Graphql/saga.js

```diff
@@ -21,6 +22,146 @@ import {
+    voting: [ApproveVote, Vote],
+    resultSetting: [ApproveSetResult, SetResult],
+    finalizing: [FinalizeResult],
+  }[phase] || [];
+  const isPending = oracle.transactions.some(({ type, status }) => pendingTypes.includes(type) && status === Pending);
+
+  const isUpcoming = phase === 'voting' && oracle.status === OracleStatus.WaitResult;
+
+  const buttonText = {
+    betting: messages.placeBet,
+    resultSetting: messages.setResult,
+    voting: messages.arbitrate,
+    finalizing: messages.finalizeResult,
+    withdrawing: messages.withdraw,
+  }[phase];
+
+  const amount = parseFloat(_.sum(oracle.amounts)).toFixed(2);
+
+  return {
+    amountLabel: phase === 'finalizing' ? `${amount} ${oracle.token}` : '',
+    url: `/oracle/${oracle.topicAddress}/${oracle.address}/${oracle.txid}`,
+    endTime: phase === 'resultSetting' ? oracle.resultSetEndTime : oracle.endTime,
+    unconfirmed: (!oracle.topicAddress && !oracle.address) || isPending,
+    isUpcoming,
+    buttonText,
+    phase,
+    ...oracle,
+  };
+});
+
+/**
+* Adds computed properties to each topic used throughout the app
+*/
+const massageTopics = (topics) => topics.map((topic) => {
+  const pendingTypes = [TransactionType.WithdrawEscrow, TransactionType.Withdraw];
+  const isPending = topic.transactions.some(({ type, status }) => pendingTypes.includes(type) && status === Pending);
+
+  const totalQTUM = parseFloat(_.sum(topic.qtumAmount).toFixed(2));
+  const totalBOT = parseFloat(_.sum(topic.botAmount).toFixed(2));
+  return {
+    ...topic,
+    amountLabel: `${totalQTUM} QTUM, ${totalBOT} BOT`,
+    url: `/topic/${topic.address}`,
+    isUpcoming: false,
+    buttonText: messages.withdraw,
+    unconfirmed: isPending,
+  };
+});
+
+
+// Send allTopics query
+export function* getAllEventsHandler() {
+  yield takeEvery(actions.GET_ALL_EVENTS, function* getAllEventsRequest(action) {
+    try {
+      const { filters, orderBy, limit, skip } = action;
+      const voteFilters = [];
+      const topicFilters = [];
+
+      // Get all votes for all your addresses
+      _.each(action.walletAddresses, ({ address }) => {
+        voteFilters.push({ voterQAddress: address });
+        topicFilters.push({ status: OracleStatus.Withdraw, creatorAddress: address });
+      });
+
+      // Filter votes
+      let votes = yield call(queryAllVotes, voteFilters);
+      votes = getUniqueVotes(votes);
+
+      // Fetch topics against votes that have the winning result index
+      _.each(votes, (vote) => {
+        topicFilters.push({ status: OracleStatus.Withdraw, address: vote.topicAddress, resultIdx: vote.optionIdx });
+      });
+      const topicsResult = yield call(queryAllTopics, topicFilters, orderBy, limit, skip);
+      const topics = _.map(topicsResult, processTopic);
+
+
+      const oraclesResult = yield call(queryAllOracles, filters, orderBy, limit, skip);
+
+      // Filter out duplicate topics
+      const oracles = _.uniqBy(oraclesResult.map(processTopic), 'txid');
```

Comment:
1. good catch
2. will do

Reply:
![image](https://user-images.githubusercontent.com/5455859/41175933-9000ab88-6b13-11e8-920d-d2bdb24da650.png)

Reply:
cool if it works 👍 but this might be related to your issue of topics not showing up. can try comparing the existing code of filtering out unique topics vs using `uniqBy`

Reply:
trying now

Reply:
That's not the problem. I just tried and it still doesn't load. I think it might be bad data. Try going here. http://localhost:3000/topic/e3181e42bd4444f6182c10386d951a98060d16d0